src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -300,14 +296,32 @@ private function ensurePhpUnrecognizedSelfClosingTagsAreXml($html)
      */
     private function ensureExistenceOfBodyElement()
     {
-        if ($this->domDocument->getElementsByTagName('body')->item(0) !== null) {
+        if ($this->getDomDocument()->getElementsByTagName('body')->item(0) !== null) {
             return;
         }
 
-        $htmlElement = $this->domDocument->getElementsByTagName('html')->item(0);
+        $htmlElement = $this->getDomDocument()->getElementsByTagName('html')->item(0);
         if ($htmlElement === null) {
             throw new \UnexpectedValueException('There is no HTML element although there should be one.', 1569930853);
         }
-        $htmlElement->appendChild($this->domDocument->createElement('body'));
+        $htmlElement->appendChild($this->getDomDocument()->createElement('body'));
+    }
+
+    /**
+     * Ensures that AbstractHtmlProcessor::$xPath has been set
+     *
+     * @throws \UnexpectedValueException if AbstractHtmlProcessor::$xPath has not been set
+     */
+    protected function ensureXPath(): \DOMXPath
```

Comment:
I think `getXPath` would be a better name – even though it may throw an exception, it still (normally) does exactly what a ‘getter’ does.  @oliverklee, what do you think?

Reply:
From the two thumbs-ups, seems we are in agreement here 😎

Reply:
implemented on rebased psalm sprint.